src/test/scala/chiselTests/PublicModuleSpec.scala

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.probe.{Probe, ProbeValue, define}
+import circt.stage.ChiselStage
+
+class PublicModuleSpec extends ChiselFlatSpec with MatchesAndOmits {
+
+  class Baz extends RawModule
+
+  class Bar extends RawModule with Public {
+    val a = IO(Output(Probe(Bool())))
+    val baz = Module(new Baz)
+
+    val b = WireInit(Bool(), true.B)
+    dontTouch(b)
+
+    define(a, ProbeValue(b))
+  }
+
+  class Foo extends RawModule {
+    val a = IO(Output(Probe(Bool())))
+    val bar = Module(new Bar)
+    define(a, bar.a)
+  }
+
+  "The main module" should "be marked public" in {
+
+    println(ChiselStage.emitCHIRRTL(new Foo))
+    println(ChiselStage.emitSystemVerilog(new Foo))
+
+    matchesAndOmits(ChiselStage.emitCHIRRTL(new Foo))(
+      "module Baz",
+      "public module Bar",
+      "module Foo"
+    )(
+      "public module Baz",
+      "public module Foo"
+    )
+
```

Comment:
The words of this test seem contradictory with the behavior, `Foo` is not being marked `public` but shouldn't it be?

Reply:
Clarified below. In emission (at this point in time for the PR and later changed), `Foo` does not have the "public" keyword. However, it's the main module, so it has different semantics in that it is treated as if it were public. It's less confusing to not do it this way and was changed in a later fixup.